File: ios/Sources/ForgerockBridgePlugin/ErrorHandler.swift
import Capacitor

enum ErrorCode: String, Error {
    case unknown_error              = "FRE000"
    case missingJourney             = "FRE024"
    case authenticateFailed         = "FRE025"
    case noAccountsRegistered       = "FRE026"
    case noOtpRegistered            = "FRE027"
    case withOutInitializedShared   = "FRE028"
    case deleteOTPFailed            = "FRE029"
    case callbackFailed             = "FRE030"
    case registerOTPFailed          = "FRE031"
    case gettingUserInfo            = "FRE033"
    case httpRequestError           = "FRE034"
    case missingParameter           = "FRE035"
    case noPendingNode              = "FRE036"
    case noQuestionFound            = "FRE037"
}

struct ErrorHandler {
    
    static func reject(_ call: CAPPluginCall, code: ErrorCode) {
        call.reject("", code.rawValue )
    }
    
    static func reject(_ call: CAPPluginCall, code: ErrorCode, message:String) {
        call.reject(message, code.rawValue )
    }
    
}

File: android/src/main/java/com/plugins/forgerockbridge/ErrorHandler.java
package com.plugins.forgerockbridge;

import com.getcapacitor.PluginCall;

public class ErrorHandler {
    public enum ErrorCode {
        UNKNOWN_ERROR("FRE000"),
        MISSING_JOURNEY("FRE024"),
        AUTHENTICATE_FAILED("FRE025"),
        NO_ACCOUNTS_REGISTERED("FRE026"),
        NO_OTP_REGISTERED("FRE027"),
        WITHOUT_INITIALIZED_SHARED("FRE028"),
        DELETE_OTP_FAILED("FRE029"),
        DELETE_OTP_MECHANISM_FAILED("FRE030"),
        CALLBACK_FAILED("FRE031"),
        REGISTER_OTP_FAILED("FRE032"),
        GETTING_USER_INFO("FRE033"),
        HTTP_REQUEST_ERROR("FRE034");

        public final String code;
        ErrorCode(String code) { this.code = code; }
    }


    public static void reject(PluginCall call, ErrorCode code) {
      call.reject("", code.code);
    }

    public static class FRException extends Exception {
      private final ErrorHandler.ErrorCode code;

      public FRException(ErrorHandler.ErrorCode code) {
          super(code.name());
          this.code = code;
      }
      
      public ErrorHandler.ErrorCode getCode() {
          return code;
      }
    }

}
